nginx/config.go

```diff
@@ -308,6 +308,9 @@ http {
 				proxy_set_header X-Request-Id $request_id;
 				proxy_set_header X-Correlation-Id $correlation_id;
 				{{ end }}
+				{{ if and $routerConfig.RequestStartHeader (or $appConfig.RequestStartHeader (eq $appConfig.RequestStartHeader nil))}}
```

Comment:
@pdomagala @felixbuenemann Can you guys take a look at this? Is this the logic we are looking for?